packages/core/src/components/graphs/area.ts

```diff
@@ -37,18 +43,51 @@ export class Area extends Component {
 				.x((d, i) => cartesianScales.getDomainValue(d, i))
 				.y0(cartesianScales.getRangeValue(0))
 				.y1((d, i) => cartesianScales.getRangeValue(d, i));
+			minAndMax = this.services.cartesianScales.getMainYScale().domain();
 		} else {
 			areaGenerator
 				.x0(cartesianScales.getRangeValue(0))
 				.x1((d, i) => cartesianScales.getRangeValue(d, i))
 				.y((d, i) => cartesianScales.getDomainValue(d, i));
+			minAndMax = this.services.cartesianScales.getMainXScale().domain();
 		}
 
 		// Update the bound data on area groups
 		const groupedData = this.model.getGroupedData();
+
+		// Should gradient style be applicable
+		const isGradientAllowed =
+			groupedData
+			&& groupedData.length === 1
+			&& Tools.getProperty(this.model.getOptions(), "gradientEnabled");
+
 		const areas = svg
 			.selectAll("path.area")
 			.data(groupedData, (group) => group.name);
+		if (!this.parent.selectAll("defs linearGradient").empty()) {
+			this.parent
+				.selectAll("defs linearGradient")
+				.each(function() {
+					this.parentNode.remove();
+				});
+		}
+		if (isGradientAllowed) {
+			groupedData.forEach((dataset) => {
+				GradientUtils.appendLinearGradient(
+					this.parent,
+					dataset.name.replace(" ", "") + "_" + this.gradient_id,
+					"0%",
+					"0%",
+					"0%",
+					"100%",
+					GradientUtils.getStopArray(
```

Comment:
`getStopArray` should just receive `minAndMax` instead and it'd talk to `need3Stops` internally